applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -238,7 +238,17 @@ def AdvanceInTime(self, current_time):
         return new_time
 
     def ComputeDeltaTime(self):
-        return self.settings["time_stepping"]["time_step"].GetDouble()
+        if self.settings["time_stepping"].Has("time_step"):
+            return self.settings["time_stepping"]["time_step"].GetDouble()
+        elif self.settings["time_stepping"].Has("time_step_table"):
+            current_time = self.main_model_part.ProcessInfo[KratosMultiphysics.TIME]
+            time_step_table = self.settings["time_stepping"]["time_step_table"].GetMatrix()
+            tb = KratosMultiphysics.PiecewiseLinearTable()
```

Comment:
possible improvement would be to save it and then check with `hasattr` then you wouldn't need to read it every time

Reply:
Is an small object, we can live with this...

Reply:
I am more concerned with the reading from parameters, but you are right it does not make a big difference